apps/web/src/components/MintLinkButton.tsx

```diff
@@ -0,0 +1,144 @@
+import { MINT_LINK_DISABLED_CONTRACTS } from '~/shared/utils/communities';
+import {
+  getMintUrlWithReferrer,
+  MINT_LINK_CHAIN_ENABLED,
+} from '~/shared/utils/getMintUrlWithReferrer';
+
+import { Button, ButtonProps } from './core/Button/Button';
+import VerifyNavigationPopover from './core/GalleryLink/VerifyNavigationPopover';
+import { HStack } from './core/Spacer/Stack';
+
+type Props = {
+  tokenRef: MintLinkButtonFragment$key;
+  overwriteURL?: string;
+  referrerAddress?: string;
+  size?: 'sm' | 'md';
+} & ButtonProps;
+
+export function MintLinkButton({
+  overwriteURL,
+  referrerAddress,
+  tokenRef,
+  size = 'md',
+  variant = 'primary',
+  ...props
+}: Props) {
+  const token = useFragment(
+    graphql`
+      fragment MintLinkButtonFragment on Token {
+        definition {
+          community {
+            contract {
+              mintURL
+              contractAddress {
+                chain
+                address
+              }
+            }
+          }
+        }
+      }
+    `,
+    tokenRef
+  );
+
+  const { showModal } = useModalActions();
+
+  const tokenContractAddress =
+    token?.definition?.community?.contract?.contractAddress?.address ?? '';
+  const tokenChain = token?.definition?.community?.contract?.contractAddress?.chain ?? '';
+  const { url: mintURL, provider: mintProviderType } = getMintUrlWithReferrer(
+    overwriteURL ?? token?.definition?.community?.contract?.mintURL ?? '',
+    referrerAddress ?? ''
+  );
+
+  const mintProvider: {
+    buttonText: string;
+    icon: React.ReactNode;
+  } | null = useMemo(() => {
+    if (mintProviderType === 'Zora') {
+      return {
+        buttonText: 'mint on zora',
+        icon: <ZoraLogoIcon />,
+      };
+    } else if (mintProviderType === 'MintFun') {
+      return {
+        buttonText: 'mint on mint.fun',
+        icon: <MintFunLogoIcon />,
+      };
+    } else if (mintProviderType === 'FxHash') {
+      return {
+        buttonText: 'mint on fxhash',
+        icon: <FxHashLogoIcon mode={variant === 'primary' ? 'light' : 'dark'} />,
+      };
+    } else if (mintProviderType === 'Prohibition') {
+      return {
+        buttonText: 'mint on prohibition',
+        icon: <ProhibitionLogoIcon mode={variant === 'primary' ? 'light' : 'dark'} />,
+      };
+    } else {
+      return null;
+    }
+  }, [mintProviderType, variant]);
+
+  const handleMintButtonClick = useCallback(() => {
+    showModal({
+      content: <VerifyNavigationPopover href={mintURL} eventContext={contexts.Feed} />,
+      isFullPage: false,
+      headerText: 'Leaving gallery.so?',
+    });
+  }, [mintURL, showModal]);
+
+  const arrowColor = useMemo(() => {
+    if (variant === 'primary') {
+      return colors.white;
+    }
+    return colors.black[800];
+  }, [variant]);
+
+  if (MINT_LINK_DISABLED_CONTRACTS.has(tokenContractAddress)) {
+    return null;
+  }
+
+  if (MINT_LINK_CHAIN_ENABLED.indexOf(tokenChain) < 0) {
+    return null;
+  }
```

Comment:
hypernit: could we convert `MINT_LINK_CHAIN_ENABLED` into a Set, similar to L110 `MINT_LINK_DISABLED_CONTRACTS`?

Reply:
Updated to use `Set`!